src/Workspaces/MSBuild/Core/MSBuild/MSBuildProjectLoader.Worker.cs

```diff
@@ -412,18 +412,66 @@ private IEnumerable<AnalyzerReference> ResolveAnalyzerReferences(CommandLineArgu
             {
                 string? fullPath;
 
-                if (PathUtilities.IsAbsolute(path))
-                {
-                    fullPath = FileUtilities.TryNormalizeAbsolutePath(path);
+                if (!PathUtilities.IsAbsolute(path))
+                    continue;
 
-                    if (fullPath != null && File.Exists(fullPath))
-                    {
-                        analyzerLoader.AddDependencyLocation(fullPath);
-                    }
-                }
+                fullPath = FileUtilities.TryNormalizeAbsolutePath(path);
+
+                if (fullPath != null && File.Exists(fullPath))
+                    analyzerLoader.AddDependencyLocation(fullPath);
             }
 
-            return commandLineArgs.ResolveAnalyzerReferences(analyzerLoader).Distinct(AnalyzerReferencePathComparer.Instance);
+            var analyzerReferences = commandLineArgs.ResolveAnalyzerReferences(analyzerLoader).Distinct(AnalyzerReferencePathComparer.Instance);
+
+            // Force loading of the analyzers so that we can record any load failures now.
+            foreach (var analyzerReference in analyzerReferences.OfType<AnalyzerFileReference>())
```

Comment:
This seems a bit expensive if you're not going to be actually looking at all the projects (or only looking at source). Isn't there a MEF factory this could export that can hook this up?